daphne/src/testing.rs

```diff
@@ -341,10 +348,66 @@ impl DapLeader<BearerToken> for MockAggregator {
 
     async fn get_reports(
         &self,
-        _task_id: &Id,
-        _selector: &Self::ReportSelector,
+        task_id: &Id,
+        selector: &Self::ReportSelector,
     ) -> Result<Vec<Report>, DapError> {
-        unreachable!("not implemented");
+        let task_config = self
+            .get_task_config_for(task_id)
+            .ok_or_else(|| DapError::fatal("no task found"))?;
+
+        // Calculate batch interval.
+        let batch_interval = if let Some(ref interval) = selector.batch_info {
+            if !interval.is_valid_for(task_config) {
+                return Err(DapError::fatal("invalid batch interval"));
+            }
+            interval.clone()
+        } else {
+            task_config.current_batch_window(Utc::now().timestamp() as u64)
+        };
+
+        // Do some checks on batch interval.
+        if batch_interval.start > Utc::now().timestamp() as u64 {
```

Comment:
This check would fail if the start of the batch interval is in the future, but I'm not sure that's actually what we want.

Tip: Test code should not be time sensitive, unless time sensitivity is the thing being tested. Here I'm not sure it is.

Reply:
Probably we should return an empty `Vec<Report>` even if the interval is to the future.
I'll work on this.

Reply:
Simply removing this checks returns an empty vector, which I think is expected.